### DISCUSSION TITLE
M2A: How can I reorder the collection items?

### DISCUSSION THREAD

**Discussion by m-kind** [directus/directus#15550] [Archive]
I am on Directus 9.x. In my data model I have created an M2A field. When editing the content I can append items to the collection and delete items from it. However, I cannot rearrange the order of items in the collection (e.g. by dragging them). 

In Directus the UI shows up like this:
![directus](https://user-images.githubusercontent.com/20257003/189880932-6555f350-db16-4b7d-9093-0dfdbbda6e09.png)

while e.g. in Strapi it is possible to reorder the "components" in a "container":
![strapi](https://user-images.githubusercontent.com/20257003/189881187-4c681c0b-0f9c-4d5f-97d0-c3a106d494a8.png)

How can I accomplish this in Directus? I tried to edit the raw data of the M2A field in the UI but at no avail.

Thanks in advance for your help!
Marko

**Comment by br41nslug** [comment]
To enable manual ordering you'll need to add a `sort` field in your data-model. 
![firefox_gOljCzZXA9](https://user-images.githubusercontent.com/9389634/189885375-48b9efb7-182b-48d4-96f1-bb1764ac4605.gif)

**Reply by m-kind** [reply]
Thanks a lot for this precious hint!

**Reply by shash7** [reply]
Good lord this is such a important part of M2A and its practically hidden from the rest of the documentation.

**Reply by benhaynes** [reply]
@MichelleEMadd let's get Bryant to do a video on each relationship type (setup, options, reason, etc) and then include those inline in the docs.

**Reply by bee8bit** [reply]
Hmm, this does not seem to work for me.
I've added (and removed) a sort field in the _M2A_ relationship and the hidden _collection_item_ collection several times, still no handles to grab and reorder the items. But I know they were there when I was doing another experiment.
So, could this be an issue with older revisions? Anything else that I might check to investigate further?

**Reply by bee8bit** [reply]
Ah, it was the pagination ...
Reordering does not work yet when the items are paginated.
Note: Setting _Items Per Page_ to -1 does seem to disable pagination, but deactivates manual sorting as well.

**Reply by jimzer** [reply]
I second this. The M2A feature is awesome, but the lack of ordering when you have multiple pages of items is problematic.

**Comment by tonyellow** [comment]
Can anyone tell me why sorting only works when having the layout of a O2M set to "List"? When selecting table, there is no option to order the results.

![Screenshot 2022-11-09 at 10 55 28](https://user-images.githubusercontent.com/8984917/200799952-684df388-e994-4ff5-8680-cc994245ed32.png)

**Reply by azrikahar** [reply]
> When selecting table, there is no option to order the results.

It was not included in the MVP when it's added in #12820 (More detail here: https://github.com/directus/directus/pull/12820#issuecomment-1123860133). This improvement is currently tracked in #15732.

**Reply by tonyellow** [reply]
Thanks, good to know. Keep up the good work with this amazing software.
